plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/android/KtlintPluginAndroidTest.kt

```diff
@@ -0,0 +1,114 @@
+package org.jlleitschuh.gradle.ktlint.android
+
+import net.swiftzer.semver.SemVer
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.hasOutcome
+import org.gradle.testkit.runner.TaskOutcome
+import org.gradle.util.GradleVersion
+import org.jlleitschuh.gradle.ktlint.AbstractPluginTest
+import org.jlleitschuh.gradle.ktlint.CHECK_PARENT_TASK_NAME
+import org.jlleitschuh.gradle.ktlint.testdsl.TestVersions
+import org.jlleitschuh.gradle.ktlint.testdsl.androidProjectSetup
+import org.jlleitschuh.gradle.ktlint.testdsl.build
+import org.jlleitschuh.gradle.ktlint.testdsl.getMajorJavaVersion
+import org.jlleitschuh.gradle.ktlint.testdsl.project
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.EnumSource
+
+class KtlintPluginAndroidTest : AbstractPluginTest() {
+
+    @ParameterizedTest
+    @EnumSource(AndroidTestInput::class)
+    fun `ktlint pass src java`(input: AndroidTestInput) {
+        if (input.minimumJava != null && getMajorJavaVersion() < input.minimumJava) {
+            return
+        }
+        if (input.maximumJava != null && getMajorJavaVersion() > input.maximumJava) {
+            return
+        }
```

Comment:
IIRC i think you may be looking for `assume` for things like this:

https://junit.org/junit5/docs/5.0.0/api/org/junit/jupiter/api/Assumptions.html

Reply:
ooo yeah that looks much better. Ill update